org.eclipse.jdt.core/model/org/eclipse/jdt/core/JavaCore.java

```diff
@@ -3396,6 +3406,7 @@ public static boolean isSupportedJavaVersion(String version) {
 	 * @since 3.39
 	 */
 	public static boolean isJavaSourceVersionSupportedByCompiler(String version) {
+		//TODO: drop this again and just mention the use-case in the doc of getAllJavaSourceVersionsSupportedByCompiler()?
```

Comment:
I would not drop it, I believe it is easier to read that compared to `getAllJavaVersionsSupportedByCompiler().contains(someVersion)` and also less error prone, as `contains(Object)` allows to pass everything inside.
So please remove TODO.

Reply:
Understand.
And maybe if we remove the `get` prefix?
I would find
`JavaCore.allJavaVersionsSupportedByCompiler().contains(someVersion)` readable. And usually either Linters or even the compiler warn if the type is unrelated (IIRC).

But it's just a suggestion, if you still prefer to keep it, I'm fine.